### PR TITLE
[AF-918] Fix SVG sanitisation in ZAT Validate to work with Zendesk Garden assets

### PR DESCRIPTION
**Pull request by fvzzy**
:v:

/cc @zendesk/vegemite

### Description
Whitelists some elements and properties so the (valid) SVG assets compiled by Zendesk Garden don't trigger a sanitisation rewrite.

### References
* JIRA: https://zendesk.atlassian.net/browse/AF-918
* Z1: https://support.zendesk.com/agent/tickets/3046110

### Risks
* [low] exposes vulnerabilities in additionally allowed elements/properties.

### CODE REVIEW

File: lib/zendesk_apps_support/validations/svg.rb [hunk 1]
```diff
@@ -4,33 +4,38 @@
...
       # but is included in many of the test apps used in fixtures for tests in ZAM, ZAT etc.
-      @remove_enable_background = Loofah::Scrubber.new do |node|
-        match_pattern = Regexp.new("enable-background:.*?(\;|\z)")
+      Loofah::HTML5::WhiteList::ACCEPTABLE_CSS_PROPERTIES.add 'enable-background'
+
+      @strip_declaration = Loofah::Scrubber.new do |node|
+        node.remove if node.name == 'xml' && node.children.empty?
+      end
+
+      @strip_spaces_between_css_attrs = Loofah::Scrubber.new do |node|
```
Comment (danielbreves) [inline lib/zendesk_apps_support/validations/svg.rb]: Why do we do this?

File: lib/zendesk_apps_support/validations/svg.rb [hunk 2]
```diff
@@ -4,33 +4,38 @@
...
+        match_pattern = Regexp.new(/\;\s+/)
         if node.name == 'svg' && node['style']
-          node['style'] = node['style'].gsub(match_pattern, '')
-          node.attributes['style'].remove if node['style'].empty?
+          node['style'] = node['style'].gsub(match_pattern, ';')
         end
       end
 
+      @empty_malformed_markup = Loofah::Scrubber.new do |node|
+        node.next.remove while node.name == 'svg' && node.next
```
Comment (danielbreves) [inline lib/zendesk_apps_support/validations/svg.rb]: Is this because we can only support one svg node?

File: spec/validations/svg_spec.rb [hunk 3]
```diff
@@ -2,11 +2,19 @@
 require 'spec_helper'
 
 describe ZendeskAppsSupport::Validations::Svg do
-  let(:clean_markup) do
+  clean_markup = [
```
Comment (danielbreves) [inline spec/validations/svg_spec.rb]: Could we define this within the 'clean markup' context for consistency?

File: spec/validations/svg_spec.rb [hunk 4]
```diff
@@ -100,7 +110,7 @@
       let(:markup) { markup }
       it 'sanitises questionable markup and notifies the user that the offending svgs were modified' do
         errors = ZendeskAppsSupport::Validations::Svg.call(package)
-        expect(IO).to have_received(:write).with(svg.relative_path, clean_markup)
+        expect(IO).to have_received(:write).with(svg.relative_path, clean_markup[0])
```
Comment (danielbreves) [inline spec/validations/svg_spec.rb]: It doesn't seem clear that the first `clean_markup` is a sanitised version of these. Could we redefine it here to be explicit?

File: lib/zendesk_apps_support/validations/svg.rb (hunk 1 above)
Comment (fvzzy) [reply lib/zendesk_apps_support/validations/svg.rb]: Loofah condenses these spaces in it's optimisation of the markup, so stripping them before sanitising ensures we don't unnecessarily trigger a rewrite.* 

*(This is if you're referring to the `@strip_spaces_between_css_attrs` - I think you are, but its hard to tell)

File: lib/zendesk_apps_support/validations/svg.rb (hunk 2 above)
Comment (fvzzy) [reply lib/zendesk_apps_support/validations/svg.rb]: Yep, sort of. I think it was also so we didn't have subsequent tags following/outside the main <svg>. I do remember adding this to make the tests at https://github.com/zendesk/zendesk_apps_support/blob/cd87351ed0bff97347c655e59a3e512b5ba572b6/spec/validations/svg_spec.rb#L120 make sense

File: lib/zendesk_apps_support/validations/svg.rb (hunk 1 above)
Comment (danielbreves) [reply lib/zendesk_apps_support/validations/svg.rb]: Could you add a comment please?

**Comment by fvzzy** [comment]
Example steps to QA:
1. Checkout https://github.com/zopim/chat-dynamic-content-app
2. `webpack` to build assets into `dist`
3. Inside local ZAT, temporarily point ZAS dependency to this branch
4. Run `zat validate` using local ZAT inside `chat-dynamic-content-app`
5. Validation should pass without triggering a rewrite on `index.svg`

**Pull request merged**